WordPress/gutenberg
# Post Title Block: add typography formatting options

## Description

Adds text formatting options to the Post Title block using experimental block supports declarations.

### Summary of changes

- Adds font weight, font style, and text transform typography controls to the Post Title block.

Partially addresses #31117, but leaves out text decoration, for now

## How has this been tested?

- Install and activate the [tt1-blocks theme](https://github.com/WordPress/theme-experiments/tree/e80d8d5c59859fef449d83d428ddeca7906ad830/tt1-blocks)
- Open a template in the Site Editor that uses a Post Title block, such as Index or Single
- Test the additional typography settings for the block. Note that they apply to the entire block contents, they are _not_ inline controls
- Make sure they work as expected with "Make title a link" on and off

## Screenshots 

<img width="280" alt="Screen Shot 2021-09-14 at 14 38 55" src="https://user-images.githubusercontent.com/1699996/133323132-44ba50b3-8970-432e-b913-3344f711ccd7.png">


## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
